src/operator/multicast.ts

```diff
@@ -26,11 +33,35 @@ export function multicast<T>(subjectOrSubjectFactory: Subject<T> | (() => Subjec
       return <Subject<T>>subjectOrSubjectFactory;
     };
   }
-  return new ConnectableObservable(this, subjectFactory);
+
+  if (selector) {
+    return new MulticastObservable(this, subjectFactory, selector);
+  } else {
+    return new ConnectableObservable(this, subjectFactory);
+  }
+}
+
+class MulticastObservable<T> extends Observable<T> {
```

Comment:
Also, why not give it a static `create` function? It would then be possible to use it like:

```js
Observable.multicast(source, () => new Subject(), 
  (shared) => shared.map(x => x + '!')
                    .merge(shared.map(x => x + '?')));
```